### PR TITLE
mock API

### PR DESCRIPTION
**Pull request by danieltolmon**
feature #13 

Mocking API using mirageJs

### CODE REVIEW

**Comment by danieltolmon** [comment]
The api is mocked using mirage. I used a db.json where I store all the data needed and a folder `routes` where I have grouped all the routes organised by models

**Review by danielgek — APPROVED** [review]
LGTM 🔥

File: src/app/utils/globals-util.ts [hunk 1]
```diff
@@ -8,8 +8,10 @@ export class RackhdLocalStorage {
   }
 
   static getBaseUrl(): string {
-    return (RackhdLocalStorage.isSecured() ? 'https://' : 'http://') +
-      window.localStorage.getItem('rackhd.northboundApi');
+
```
Comment (asantos00) [inline src/app/utils/globals-util.ts]: Why the empty lines?

File: src/miragejs/routes/catalogs.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export default function catalogsRoutes(server) {
+  server.get("/catalogs", () => server.db.catalogs);
+  server.get("nodes/:id/catalogs/dmi", (schema, request) => {
+    return server.db.catalogs.filter(catalog => {
+      if (catalog.node === `/api/2.0/nodes/${request.params.id}`) {
+        return catalog.source === "dmi";
+      } else {
```
Comment (asantos00) [inline src/miragejs/routes/catalogs.js]: remove this `else` and just `return false`

File: src/miragejs/routes/configs.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    server.post("/config", (schema, request) => {
+      let attrs = JSON.parse(request.requestBody);
+    
+      return schema.configs.create(attrs);
+    });
+
+    server.patch("/config", (schema, request) => {
+      let attrs = JSON.parse(request.requestBody);
+      console.log(attrs)
```
Comment (asantos00) [inline src/miragejs/routes/configs.js]: Remove `console.log`

File: src/miragejs/routes/configs.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (asantos00) [inline src/miragejs/routes/configs.js:12]: Empty lines

File: src/miragejs/routes/files.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export default function configRoutes(server){
+    server.get("/files", () => server.db.files);
+
+    server.delete("/filess/:id", (schema, request) => {
```
Comment (asantos00) [inline src/miragejs/routes/files.js]: This is not used

File: src/miragejs/routes/files.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+      let id = request.params.id;
+      return schema.filess.find(id).destroy();
+    });
+    server.post("/filess", (schema, request) => {
```
Comment (asantos00) [inline src/miragejs/routes/files.js]: This is not used

File: src/miragejs/server.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      template: Model,
+      graph: Model,
+      catalog: Model,
+      config: Model,
+      file: Model,
+      workflow: Model
+    }
+  });
+
+  server.routes = routes(server);
```
Comment (asantos00) [inline src/miragejs/server.js]: you can also pass the `routes` as a parameter to the `new Server`, but this works so 👍 .

If the `baseURL` and the `namespace` are always the same, I'd pass them as a parameter to the `new Server` too. It makes it a little bit clearer

File: src/miragejs/routes/tags.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export default function tagsRoutes(server){
+    server.get("/nodes/:id/tags", (schema, request) => {
+      let id = request.params.id;
```
Comment (asantos00) [inline src/miragejs/routes/tags.js:3]: Why do you get the `id` if you're returning `[]`?

File: src/miragejs/routes/profiles.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+export default function profilesRoutes(server){
+  server.get("/profiles/metadata", () => server.db.profiles);
+  server.post("/profiles", (schema, request) => {
+    let attrs = JSON.parse(request.requestBody);
+
+    return schema.obms.create(attrs);
+  });
+
```
Comment (asantos00) [inline src/miragejs/routes/profiles.js]: empty line

File: src/miragejs/routes/index.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import filesRoutes from "./files";
+import catalogsRoutes from "./catalogs";
+import workflowsRoutes from "./workflows";
+import tagsRoutes from "./tags";
+import tasksRoutes from "./tasks";
+
+export function routes(server) {
+  server.urlPrefix = "http://127.0.0.1:9090";
+  server.namespace = "/api/2.0";
+
```
Comment (asantos00) [inline src/miragejs/routes/index.js:18]: to make this shorter, you can do it like

```js
[
  nodesRoutes,
  obmsRoutes,
  // all the others
].forEach(routes => routes(server));


```

**Review by asantos00 — APPROVED** [review]
👍 Good job, just a couple of fixes

File: src/miragejs/routes/index.js (hunk 10 above)
Comment (danieltolmon) [reply src/miragejs/routes/index.js:18]: Good one!

File: src/miragejs/server.js (hunk 7 above)
Comment (danieltolmon) [reply src/miragejs/server.js]: Yeap, nice! I did the same with the data base, I think it also looks better inside the server

File: src/miragejs/routes/tags.js (hunk 8 above)
Comment (danieltolmon) [reply src/miragejs/routes/tags.js:3]: Solved, actually should be something like this: `return schema.skus.find(id)? schema.skus.find(id) : []`

**Review by danieltolmon — COMMENTED** [review]
I have reviewed all the points @asantos00 commented and I separate the db in different folders. All the changes are pushed in the following commit

**Review by asantos00 — APPROVED** [review]

**Pull request merged**